Source/WebCore/platform/graphics/mac/controls/MenuListButtonMac.mm

```diff
@@ -159,50 +162,38 @@ static void drawMenuListBackground(GraphicsContext& context, const FloatRect& re
 
     float leftEdge;
     if (isRightToLeft)
-        leftEdge = bounds.x() + arrowPaddingAfter * style.zoomFactor;
+        leftEdge = logicalBounds.x() + arrowPaddingAfter * style.zoomFactor;
     else
-        leftEdge = bounds.maxX() - arrowPaddingAfter * style.zoomFactor - arrowWidth;
+        leftEdge = logicalBounds.maxX() - arrowPaddingAfter * style.zoomFactor - arrowWidth;
 
     GraphicsContextStateSaver stateSaver(context);
 
     context.setFillColor(style.textColor);
     context.setStrokeStyle(StrokeStyle::NoStroke);
 
-    // Draw the top arrow
-    Vector<FloatPoint> arrow1 = {
+    Vector<FloatPoint> topArrow = {
         { leftEdge, centerY - spaceBetweenArrows / 2.0f },
         { leftEdge + arrowWidth, centerY - spaceBetweenArrows / 2.0f },
         { leftEdge + arrowWidth / 2.0f, centerY - spaceBetweenArrows / 2.0f - arrowHeight }
     };
-    context.fillPath(Path(arrow1));
 
-    // Draw the bottom arrow
-    Vector<FloatPoint> arrow2 = {
+    Vector<FloatPoint> bottomArrow = {
         { leftEdge, centerY + spaceBetweenArrows / 2.0f },
         { leftEdge + arrowWidth, centerY + spaceBetweenArrows / 2.0f },
         { leftEdge + arrowWidth / 2.0f, centerY + spaceBetweenArrows / 2.0f + arrowHeight }
     };
-    context.fillPath(Path(arrow2));
 
-    constexpr auto leftSeparatorColor = Color::black.colorWithAlphaByte(40);
-    constexpr auto rightSeparatorColor = Color::white.colorWithAlphaByte(40);
+    if (isVerticalWritingMode) {
+        auto transposePoint = [](const FloatPoint& point) {
+            return point.transposedPoint();
+        };
 
-    // FIXME: Should the separator thickness and space be scaled up by fontScale?
-    int separatorSpace = 2; // Deliberately ignores zoom since it looks nicer if it stays thin.
-    int leftEdgeOfSeparator;
-    if (isRightToLeft)
-        leftEdgeOfSeparator = static_cast<int>(roundf(leftEdge + arrowWidth + arrowPaddingBefore * style.zoomFactor));
-    else
-        leftEdgeOfSeparator = static_cast<int>(roundf(leftEdge - arrowPaddingBefore * style.zoomFactor));
-
-    // Draw the separator to the left of the arrows
-    context.setStrokeThickness(1); // Deliberately ignores zoom since it looks nicer if it stays thin.
-    context.setStrokeStyle(StrokeStyle::SolidStroke);
-    context.setStrokeColor(leftSeparatorColor);
-    context.drawLine(IntPoint(leftEdgeOfSeparator, bounds.y()), IntPoint(leftEdgeOfSeparator, bounds.maxY()));
+        topArrow = topArrow.map(transposePoint);
+        bottomArrow = bottomArrow.map(transposePoint);
+    }
```

Comment:
Is this rotating the arrow? If it is, is it correct to rotate it? Does the popup appear where the arrow points?

Reply:
Would be good to add a comment regardless.

Reply:
Yes, this rotates the arrows – I've added a comment. 

The positioning of the menu isn't perfect, but it does appear to the side of the control (which is what we want) rather than above or below, hence the rotation.